modin/config/envvars.py

```diff
@@ -0,0 +1,224 @@
+        if cls.choices:
+            help += f" (valid examples are: {', '.join(str(c) for c in cls.choices)})"  # pylint: disable=not-an-iterable
+        return help
+
+
+class IsDebug(EnvironmentVariable, type=bool):
+    """
+    Forces Modin engine to be "Python" unless specified by $MODIN_ENGINE
+    """
+
+    varname = "MODIN_DEBUG"
+
+
+class Engine(EnvironmentVariable, type=str):
+    """
+    Distribution engine to run queries by
+    """
+
+    varname = "MODIN_ENGINE"
+    choices = ("Ray", "Dask", "Python")
+
+    @classmethod
+    def _get_default(cls):
+        if IsDebug.get():
+            return "Python"
+        try:
+            import ray
+
+        except ImportError:
+            pass
+        else:
+            if version.parse(ray.__version__) < version.parse("1.0.0"):
+                raise ImportError(
+                    "Please `pip install modin[ray]` to install compatible Ray version."
+                )
+            return "Ray"
+        try:
+            import dask
+            import distributed
+
+        except ImportError:
+            raise ImportError(
+                "Please `pip install modin[ray]` or `modin[dask]` to install an engine"
+            )
+        if version.parse(dask.__version__) < version.parse("2.1.0") or version.parse(
+            distributed.__version__
+        ) < version.parse("2.3.2"):
+            raise ImportError(
+                "Please `pip install modin[dask]` to install compatible Dask version."
+            )
+        return "Dask"
+
+
+class Backend(EnvironmentVariable, type=str):
+    """
+    Engine running on a single node of distribution
+    """
+
+    varname = "MODIN_BACKEND"
+    default = "Pandas"
+    choices = ("Pandas", "OmniSci", "Pyarrow")
+
+
+class IsExperimental(EnvironmentVariable, type=bool):
+    """
+    Turns on experimental features
+    """
+
+    varname = "MODIN_EXPERIMENTAL"
+
+
+class IsRayCluster(EnvironmentVariable, type=bool):
+    """
+    True if Modin is running on pre-initialized Ray cluster
+    """
+
+    varname = "MODIN_RAY_CLUSTER"
+
+
+class RayRedisAddress(EnvironmentVariable, type=str):
+    """
+    What Redis address to connect to when running in Ray cluster
+    """
+
+    varname = "MODIN_REDIS_ADDRESS"
+
+
+class CpuCount(EnvironmentVariable, type=int):
+    """
+    How may CPU cores to utilize across the whole distribution
+    """
+
+    varname = "MODIN_CPUS"
+
+    @classmethod
+    def _get_default(cls):
+        import multiprocessing
+
+        return multiprocessing.cpu_count()
+
+
+class Memory(EnvironmentVariable, type=int):
+    """
+    How much memory give to each Ray worker (in bytes)
+    """
+
+    varname = "MODIN_MEMORY"
+
+
+class RayPlasmaDir(EnvironmentVariable, type=str):
+    """
+    Path to Plasma storage for Ray
+    """
+
+    varname = "MODIN_ON_RAY_PLASMA_DIR"
+
+
+class IsOutOfCore(EnvironmentVariable, type=bool):
```

Comment:
Missing class-level docstring

Reply:
I know, but I have no clue what this variable does :) @devin-petersohn could you pls help?

Reply:
Maybe this can be useful https://modin.readthedocs.io/en/latest/out_of_core.html

Reply:
nice find, thx, docstring added